api/envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto

```diff
@@ -765,6 +765,21 @@ message HttpConnectionManager {
   // If this config is set, the Proxy-Status HTTP response header field is
   // populated. By default, it is not.
   ProxyStatusConfig proxy_status_config = 49;
+
+  // Configuration options for Unified Header Validation (UHV).
+  // UHV is an extensible mechanism for checking validity of HTTP requests as well as providing
+  // normalization for request attributes, such as URI path.
+  // The default UHV checks the following:
+  // 1. H/1 header map validity according to `RFC 7230 section 3.2<https://tools.ietf.org/html/rfc7230#section-3.2>`_
+  // 2. Syntax of H/1 request target URI and response status
+  // 3. H/2 header map validity according to `RFC 7540 section 8.1.2<https://tools.ietf.org/html/rfc7540#section-8.1.2`_
+  // 4. Syntax of H/2 pseudo headers
+  // 5. Syntax of Content-Length and Transfer-Encoding
+  // 6. Validation of H/1 requests with both Content-Length and Transfer-Encoding headers
+  // 7. Normalization of the URI path
+  // [#not-implemented-hide:]
+  // [#extension-category: envoy.http.unified_header_validators]
+  config.core.v3.TypedExtensionConfig typed_unified_header_validation_config = 50;
```

Comment:
Is one config sufficient here?
I want to make sure we have options for @AndresGuedez case where we're going to need to branch either on VIP or header.
Would the expectation be for that use casewe have a custom root proto which has 2 UnifiedHeaderValidatorConfigs and matchers for which to use when?

Reply:
Matching on header will not work since UHV is called before the entire header map is available. We could add matchers on L4 attributes though, but it will not help changing behavior per request.

My current thinking is that L1 need to add a header at the beginning of the header map to indicate L2 that it needs to use  compatibility mode for the request. I'm not 100% sure it would work. We can discuss internally to see if something else is needed.